tests/components/buienradar/test_camera.py

```diff
@@ -1,34 +1,63 @@
 """The tests for generic camera component."""
 import asyncio
 from contextlib import suppress
+import copy
 
 from aiohttp.client_exceptions import ClientResponseError
 
-from homeassistant.const import HTTP_INTERNAL_SERVER_ERROR
-from homeassistant.setup import async_setup_component
+from homeassistant.components.buienradar.const import CONF_COUNTRY, CONF_DELTA, DOMAIN
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    HTTP_INTERNAL_SERVER_ERROR,
+)
+from homeassistant.helpers.entity_registry import async_get
 from homeassistant.util import dt as dt_util
 
+from tests.common import MockConfigEntry
+
 # An infinitesimally small time-delta.
 EPSILON_DELTA = 0.0000000001
 
+TEST_LATITUDE = 51.5288504
+TEST_LONGITUDE = 5.4002156
+
+TEST_CFG_DATA = {CONF_LATITUDE: TEST_LATITUDE, CONF_LONGITUDE: TEST_LONGITUDE}
+
+
+def radar_map_url(country_code: str = "NL") -> str:
+    """Build map URL."""
+    return f"https://api.buienradar.nl/image/1.0/RadarMap{country_code}?w=700&h=700"
+
+
+async def _setup_config_entry(hass, entry):
+    entity_registry = async_get(hass)
+    entity_registry.async_get_or_create(
+        domain="camera",
+        platform="buienradar",
+        unique_id=f"{TEST_LATITUDE:2.6f}{TEST_LONGITUDE:2.6f}",
+        config_entry=entry,
+        original_name="Buienradar",
+    )
+    await hass.async_block_till_done()
 
-def radar_map_url(dim: int = 512, country_code: str = "NL") -> str:
-    """Build map url, defaulting to 512 wide (as in component)."""
-    return f"https://api.buienradar.nl/image/1.0/RadarMap{country_code}?w={dim}&h={dim}"
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
 
 
 async def test_fetching_url_and_caching(aioclient_mock, hass, hass_client):
     """Test that it fetches the given url."""
     aioclient_mock.get(radar_map_url(), text="hello world")
```

Comment:
@MartinHjelmare Thinking about your review comments, in this test we patch the radar url, but not the sensor/weather url. Wouldn't that be wrong as well? It's setting up the ConfigEntry, so also weather and sensor.

Reply:
The aioclient_mock will patch the session so there will be no real requests going out.